osu.Game/Screens/Select/Carousel/CarouselBeatmapSet.cs

```diff
@@ -129,7 +129,19 @@ private int compareUsingAggregateMax(CarouselBeatmapSet other, Func<BeatmapInfo,
         public override void Filter(FilterCriteria criteria)
         {
             base.Filter(criteria);
-            Filtered.Value = Items.All(i => i.Filtered.Value);
+            bool match = Items.All(i => i.Filtered.Value);
+
+            if (BeatmapSet?.Equals(criteria.SelectedBeatmapSet) == true)
+            {
+                // only check ruleset equality or convertability for selected beatmap
+                Filtered.Value = !match;
+                return;
+            }
+
+            match &= criteria.Sort != SortMode.DateRanked || BeatmapSet?.DateRanked != null;
```

Comment:
Some filters requires `BeatmapInfo` to work, so I just moved these two here.

Reply:
Yes, that was the intention.